ecl/hql/hqlgram2.cpp

```diff
@@ -12003,3 +12362,186 @@ IHqlExpression *HqlGram::doParse()
     return NULL;
 }
 
+/*The following confusion  matrix holds the cost of transpositions and substitutions for weighted-Damerau-Levenshtein Distance.
```

Comment:
This PR should be based on that one, and you should add a dependency on hql.so on stringlib.so.  The string lib plugins are part of the core platform/